src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+                }
+            }
+        }
+
+        if (sender instanceof EntityPlayer)
+        {
+            final ForgePermissionNodes forgePermissionNode = actionMenu.getForgePermissionNode();
+
+            final EntityPlayer player = (EntityPlayer) sender;
+            if (!PermissionAPI.hasPermission(player.getGameProfile(), forgePermissionNode.getNodeName(), new PlayerContext(player)))
+            {
+                // TODO: Do something if permission check fails.
+                // But we don't have permissions set up yet.
+            }
+        }
+
+        if (null != executionActionArgumentList)
+        {
+            for (final ActionArgument executionActionArgument : executionActionArgumentList)
+            {
+                if (!executionActionArgument.isValueSet())
+                {
+                    throw new CommandException(getCommandUsage(sender, executionTreeNode));
+                }
+            }
+        }
+
+        final Class<? extends IActionCommand> clazz = actionMenu.getActionCommandClass();
+        try
+        {
+            createInstanceAndExecute(server, sender, actionMenu, clazz);
+        }
+        catch (final InstantiationException | IllegalAccessException e)
+        {
+            final Logger log = LogManager.getLogger();
+            log.error("Unable to instantiate class %s for command %s ", clazz.getName(), actionMenu.getDescription(), e);
+            throw new CommandException("Unable to instantiate class " + clazz.getName() + " for command " + actionMenu.getDescription(), e);
+        }
+    }
+
+    protected void createInstanceAndExecute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu,
+            @NotNull final Class<? extends IActionCommand> clazz) throws InstantiationException, IllegalAccessException, CommandException
+    {
+        final IActionCommand actionCommand = clazz.newInstance();
+        actionCommand.execute(server, sender, actionMenu);
+    }
+
+    /**
+     * Check if the player has the permission to use commands.
+     *
+     * @param server the server to check for.
+     * @param sender the sender of the command.
+     * @return true if so.
+     */
+    @Override
+    public boolean checkPermission(final MinecraftServer server, final ICommandSender sender)
+    {
+        if (sender instanceof EntityPlayer)
+        {
+            return AbstractSingleCommand.isPlayerOpped(sender) || Configurations.gameplay.opLevelForServer <= 0;
+        }
+        return true;
+    }
+
+    private Map<String, TreeNode<Menu>> getNavigationCommands(@NotNull final TreeNode<Menu> parentTreeNode)
+    {
+        final Map<String, TreeNode<Menu>> map = new HashMap<>();
+        final List<TreeNode<Menu>> children = parentTreeNode.getChildren();
+        for (final TreeNode<Menu> treeNode : children)
+        {
+            map.put(treeNode.getData().getMenuItemName().toLowerCase(), treeNode);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

Reply:
this